File: src/main/kotlin/org/ozwillo/dcimporter/config/marchesecurise/MarcheSecuriseRabbitMQ.java
package org.ozwillo.dcimporter.config.marchesecurise;

import org.ozwillo.dcimporter.model.datacore.DCBusinessResourceLight;
import org.ozwillo.dcimporter.service.marchesecurise.rabbitMQ.ReceiverMS;
import org.ozwillo.dcimporter.service.marchesecurise.rabbitMQ.SenderMS;
import org.springframework.amqp.core.Binding;
import org.springframework.amqp.core.BindingBuilder;
import org.springframework.amqp.core.Queue;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.springframework.amqp.core.TopicExchange;

@Configuration
public class MarcheSecuriseRabbitMQ {
    private final String EXCHANGER_NAME = "dcimporter";

    @Bean
    public TopicExchange topic(){
        return new TopicExchange(EXCHANGER_NAME);
    }

    private static class ReceiverConfig{

        private final String QUEUE_MS_NAME = "marchesecurise";
        private final String KEY = "consultation.#";

        public String getKey(String siret, String baseKey){
            return baseKey +"." + siret + ".#";
        }

        @Bean
        public ReceiverMS receiver(){
            return new ReceiverMS();
        }

        @Bean
        public Queue queueMS(){
            return new Queue(QUEUE_MS_NAME);
        }

        @Bean
        public Binding bindingTestConsultation(TopicExchange topic, Queue queueMS){
            return BindingBuilder.bind(queueMS).to(topic).with(KEY);
        }

    }

    @Bean
    public SenderMS sender(){
        return new SenderMS();
    }

}

File: src/main/kotlin/org/ozwillo/dcimporter/config/marchesecurise/MarcheSecuriseRabbitMQ.kt
package org.ozwillo.dcimporter.config.marchesecurise

import org.ozwillo.dcimporter.repository.BusinessMappingRepository
import org.ozwillo.dcimporter.service.marchesecurise.rabbitMQ.ReceiverMS
import org.ozwillo.dcimporter.service.marchesecurise.rabbitMQ.SenderMS
import org.springframework.amqp.core.Binding
import org.springframework.amqp.core.BindingBuilder
import org.springframework.amqp.core.Queue
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.amqp.core.TopicExchange
import org.springframework.beans.factory.annotation.Autowired

@Configuration
class MarcheSecuriseRabbitMQ {
    private val EXCHANGER_NAME = "dcimporter"


    @Bean
    fun topic(): TopicExchange {
        return TopicExchange(EXCHANGER_NAME)
    }

    private class ReceiverConfig {

        @Autowired
        private lateinit var businessMappingRepository: BusinessMappingRepository

        private val QUEUE_MS_NAME = "marchesecurise"
        private val KEY_CONSULTATION = "consultation.#"

        @Bean
        fun receiver(): ReceiverMS {
            return ReceiverMS(businessMappingRepository)
        }

        @Bean
        fun queueMS(): Queue {
            return Queue(QUEUE_MS_NAME)
        }

        @Bean
        fun bindingConsultation(topic: TopicExchange, queueMS: Queue): Binding {
            return BindingBuilder.bind(queueMS).to(topic).with(KEY_CONSULTATION)
        }

    }

    @Bean
    fun sender(): SenderMS {
        return SenderMS()
    }

}